src/lib/app.ts

```diff
@@ -179,7 +181,11 @@ export default async function getApp(
     );
 
     if (typeof config.preRouterHook === 'function') {
-        config.preRouterHook(app, config, services, stores, db);
+        const { services: enterpriseServices, stores: enterpriseStores } =
+            config.preRouterHook(app, config, services, stores, db);
+
+        combinedServices = enterpriseServices;
+        combinedStores = enterpriseStores;
```

Comment:
Do we need to do this? I think the "beauty" of the getApp method is that it just return an app, and the preHook just modifies it

If we have to do that, I wouldn't call them enterpriseServices, I'd do something like:
```typescript
const { services: extendedServices, stores: extendedStores } =
            config.preRouterHook(app, config, services, stores, db);
```